locations/spiders/dillards.py

```diff
@@ -4,46 +4,41 @@
 from scrapy.spiders import SitemapSpider
 
 from locations.hours import OpeningHours
-from locations.items import Feature
+from locations.structured_data_spider import StructuredDataSpider
 
 
-class DillardsSpider(SitemapSpider):
+class DillardsSpider(SitemapSpider, StructuredDataSpider):
     name = "dillards"
     item_attributes = {"brand": "Dillard's", "brand_wikidata": "Q844805"}
     allowed_domains = ["www.dillards.com"]
     sitemap_urls = ("https://www.dillards.com/sitemap/sitemap_storeLocator_1.xml",)
     download_delay = 0.5
+    wanted_types = ["DepartmentStore"]
+    time_format = "%I:%M %p"
 
     sitemap_rules = [(r"https://www.dillards.com/stores/.*/.*", "parse")]
 
-    def parse(self, response):
-        if "Access Denied" in response.text:
-            return get_retry_request(response.request, spider=self, reason="throttle")
-
-        ldjson = response.xpath('//script[@type="application/ld+json"]/text()[contains(.,"DepartmentStore")]').get()
-        data = json.loads(ldjson)
-
+    def post_process_item(self, item, response, ld_data):
         script = response.xpath('//script/text()[contains(.,"__INITIAL_STATE__")]').get()
         script_data = json.decoder.JSONDecoder().raw_decode(script, script.index("{"))[0]
         lat = script_data["contentData"]["store"]["latitude"]
         lon = script_data["contentData"]["store"]["longitude"]
 
+        # TODO: Why isn't this automatically done by StructuredDataSpider?
```

Comment:
`dayOfWeek` is a dict, it could probably cleaned in pre process and dropped here entirely.

Reply:
Doesn't work anyway.
`cat dillards.geojson| jq '.features[].properties.opening_hours' | sort | uniq -c | sort -n`
```
      1 "Mo 12:00-19:00"
     27 null
    245 "Mo 11:00-20:00"
```